semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/description/SymbolDescription.kt

```diff
@@ -0,0 +1,39 @@
+package com.strumenta.kolasu.semantics.symbol.description
+
+data class SymbolDescription(
+    val identifier: String,
+    val types: List<String>,
+    val properties: Map<String, ValueDescription>
+)
+
+sealed interface ValueDescription {
+    val value: Any?
+}
+
+data class BooleanValueDescription(
+    override val value: Boolean? = null
+) : ValueDescription
+
+data class IntegerValueDescription(
+    override val value: Int? = null
+) : ValueDescription
+
+data class StringValueDescription(
+    override val value: String? = null
+) : ValueDescription
+
+data class ReferenceValueDescription(
+    override val value: SymbolDescription? = null
+) : ValueDescription
+
+data class ListValueDescription(
+    override val value: List<ValueDescription> = emptyList()
+) : ValueDescription
+
+data class SetValueDescription(
```

Comment:
We do not use set in Kolasu

Reply:
Perhaps we could keep it for future developments?

Reply:
I would rather not because there is a lot of code that depends on the assumption that we restrict what goes into Nodes (and I see Symbols as a "view" of nodes).

For example: transformers, Json serialization, Xml serialization, export to LionWeb, starlasu tools, IDEA plugin, compiler plugins, etc.

Supporting more types in all these aspects would be a lot of work and it would make the codebase significantly more complex. It would means also making the library more difficult to learn and to extend.

Supporting Sets or Maps in only a subset of the functionalities of Kolasu, could lead to errors while using the library

Reply:
I agree, I will remove sets and maps value descriptions then

Reply:
Thank you, I will write down in the wiki a page about what can go in the AST and what cannot go in it, and the reasons for it because I realize I never actually wrote about thi